File: Task1118/Sources/Task1118/validation.swift
import Foundation
#if DEVELOPING
import TestData

let data1 = TestData(name: "Data 1", text: """
3 5
80 20 15 50 30
25 40 95 60 85
90 10 30 70 50
""", expected: """
110
""")

let data2 = TestData(name: "Data 2", text: """
1 10
529 463 931 247 385 410 629 666 849 425
""", expected: """
0
""")

let data3 = TestData(name: "Data 3", text: """
5 6
1 4 1 5 9 2
6 5 3 5 8 9
7 9 3 2 3 8
4 6 2 6 4 3
3 8 3 2 7 9
""", expected: """
32
""")

let data4 = TestData(name: "Data 4", text: """
""", expected: """
""")

let data5 = TestData(name: "Data 5", text: """
""", expected: """
""")

let validationList = [
    data1,
    data2,
    data3,
//    data4,
//    data5,
]

// ----------------------------------------------------------
func print(_ items: Any...) {
    TestData.write(items)
}

func validate(_ main: (() -> String) -> Void) {
    validationList.forEach { $0.execute(main) }
}
// ----------------------------------------------------------

#endif

File: Task306/src/Validation.kt
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.PrintStream

fun main() {
    val data1 = TestData("Data 1", """
        3 2
        2 4 7
    """.trimIndent(), """
        11
    """.trimIndent())

    val data2 = TestData("Data 2", """
        8 4
        3 1 4 1 5 9 2 6
    """.trimIndent(), """
        15
    """.trimIndent())

    val data3 = TestData("Data 3", """
    """.trimIndent(), """
    """.trimIndent())

    val data4 = TestData("Data 4", """
    """.trimIndent(), """
    """.trimIndent())

    val data5 = TestData("Data 5", """
    """.trimIndent(), """
    """.trimIndent())

    val validationList = listOf(
        data1,
        data2,
//        data3,
//        data4,
//        data5,
    )
    
    println()
    validationList.forEach { it.execute(::solve) }
    println("--------------------------------")
    println()
}

class TestData(
    private val name: String,
    private val data: String,
    private val expected: String,
) {
    fun execute(block: () -> Unit) {
        println("--- [$name] -------------------")
        val originalIn = System.`in`
        val originalOut = System.out
        val actual = ByteArrayInputStream(data.toByteArray()).use { inputStream ->
            ByteArrayOutputStream().use { outputStream ->
                System.setIn(inputStream)
                try {
                    System.setOut(PrintStream(outputStream))
                    try {
                        block()
                        outputStream.toString()
                    } finally {
                        System.setOut(originalOut)
                    }
                } finally {
                    System.setIn(originalIn)
                }
            }
        }
        print(actual)
        if (actual[actual.length - 1] != '\n') {
            println()
        }

        if (expected.isNotEmpty()) {        
            if (actual.trim() == expected.trim()) {
                println("✅ OK")
            } else {
                println("❌ NG")
            }
        }
    }
}
